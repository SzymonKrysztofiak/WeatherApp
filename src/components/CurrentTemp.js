import React from "react";
import PropTypes from "prop-types";

const CurrentTemp = props => (
    <div className="results-container__temp">
        {props.temperature && <p>{props.temperature}&deg;C</p>}
    </div>
);

CurrentTemp.propTypes = {
    temperature: PropTypes.string
};

export default CurrentTemp;
