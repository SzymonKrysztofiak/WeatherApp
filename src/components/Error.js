import React from "react";
import PropTypes from "prop-types";

const Error = props => (
    <div className="result-container__error">
        {props.errorMessage && (
            <p>
                <strong>{props.errorMessage}</strong>
            </p>
        )}
    </div>
);

Error.propTypes = {
    errorMessage: PropTypes.string
};

export default Error;
