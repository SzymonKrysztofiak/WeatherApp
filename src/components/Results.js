import React from "react";

const Results = props => (
    <div className="results-container__rest">
        {props.date && (
            <p>
                <strong>Date: </strong>
                {props.date}
            </p>
        )}
        {props.city &&
            props.country && (
                <p>
                    <strong>Location: </strong>
                    {props.city}, {props.country}
                </p>
            )}
        {props.description && (
            <p>
                <strong>Description: </strong>
                {props.description}
            </p>
        )}
    </div>
);

export default Results;
