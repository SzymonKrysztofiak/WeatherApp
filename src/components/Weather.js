import React from "react";
import Loader from "./Loader";

const Weather = props => (
    <div className="results-container">
        {props.isLoaded ? null : <Loader />}
        <div className="results-container__temp">
            {props.temperature && <p>{props.temperature}&deg;C</p>}
        </div>
        <div className="results-container__rest">
            {props.date && (
                <p>
                    <strong>Date: </strong>
                    {props.date}
                </p>
            )}
            {props.city &&
                props.country && (
                    <p>
                        <strong>Location: </strong>
                        {props.city}, {props.country}
                    </p>
                )}
            {props.description && (
                <p>
                    <strong>Description: </strong>
                    {props.description}
                </p>
            )}
        </div>
        <div className="result-container__error">
            {props.errorMessage && (
                <p>
                    <strong>{props.errorMessage}</strong>
                </p>
            )}
        </div>
    </div>
);

export default Weather;
