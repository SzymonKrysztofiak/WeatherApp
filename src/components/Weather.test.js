import Weather from "./Weather";
import CurrentTemp from "./CurrentTemp";
import Results from "./Results";
import Error from "./Error";

it("renders without crashing", () => {
    const wrapper = shallow(<Weather />);
    expect(wrapper).toMatchSnapshot();
});

it("includes Results", () => {
    const wrapper = shallow(<Weather />);
    expect(wrapper.containsMatchingElement(<Results />)).toEqual(true);
});
it("includes Error", () => {
    const wrapper = shallow(<Weather />);
    expect(wrapper.containsMatchingElement(<Error />)).toEqual(true);
});
