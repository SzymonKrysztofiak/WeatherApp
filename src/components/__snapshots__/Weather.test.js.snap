// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="results-container"
  >
    <Loader />
    <CurrentTemp />
    <Results />
    <Error />
  </div>
</div>
`;
